ocaml/xapi/xapi_host_driver.ml

```diff
@@ -85,6 +87,160 @@ let reload_driver_selection ~__context ~host =
   with Forkhelpers.Spawn_internal_error (err, _, _) ->
     internal_error "Failed to reload driver selection (%s): %s" __FUNCTION__ err
 
+(* Versions are compared in segments separated by punctuation characters.
+   If both segments are integers, integer comparison is used, if at least
+   one of the segments is not an integer, lexicographic comparison is used.
+   Any segment compares larger against an empty segment.
+   Thus, "1.2.3" > "1.2";
+         "1.2.3" > "1.2."
+         "1.2.3" < "1.2.fix"
+         "1.2.3" < "1.2.3fix"
+*)
+module Version = struct
+  let compare a b =
+    let split str = Str.(split_delim (regexp "[.,:;+_-]") str) in
```

Comment:
Is `Str` (thread) safe to use? I can never remember, but always get worried when I see it... :/

Reply:
I think this particular function should be thread-safe, but @edwintorok would know better. Have the issues with `Re` submodules we thought were thread-safe but ended up not being thread-safe been resolved?